packages/integrations/cloudflare/src/index.ts

```diff
@@ -115,6 +116,42 @@ export default function createIntegration(args?: Options): AstroIntegration {
 					await fs.promises.mkdir(functionsUrl, { recursive: true });
 					const directoryUrl = new URL('[[path]].js', functionsUrl);
 					await fs.promises.rename(entryUrl, directoryUrl);
+
+					// this creates a _routes.json, in case there is none present
+					// to enable cloudflare to handle those files directly (without calling the worker)
+					if (
+						!(await fs.promises
+							.stat(new URL('./_routes.json', _buildConfig.client))
+							.then((stat) => stat.isFile())
+							.catch(() => false))
+					) {
+						const specialFiles = ['_headers', '_redirects'],
+							staticFiles = (
+								await glob(`${fileURLToPath(_buildConfig.client)}/**/*`, {
+									cwd: fileURLToPath(_buildConfig.client),
+									filesOnly: true,
+								})
+							).filter((file) => specialFiles.indexOf(file) < 0);
+
+						await fs.promises.writeFile(
+							new URL('./_routes.json', _buildConfig.client),
+							JSON.stringify(
+								{
+									version: 1,
+									include: ['/*'],
+									exclude: staticFiles.map((file) => `/${file}`),
+								},
+								null,
+								2
+							)
+						);
+
+						// move the files onto the root
+						await fs.promises.cp(_buildConfig.client, _buildConfig.server, {
+							recursive: true,
+						});
+						await fs.promises.rm(_buildConfig.client, { recursive: true, force: true });
+					}
```

Comment:
Could we do the client build directly on the root instead to prevent these fs operations?

Reply:
I have rewritten the way the assets are build to get around this issue, therefore the `base` config is now considered during build and proxing the asset requests is not needed any longer (`app.removeBase`).
This was necessary to enable static asset hosting on pages, which is now the default.